sycl/include/sycl/ext/oneapi/experimental/graph.hpp

```diff
@@ -239,6 +263,9 @@ class __SYCL_EXPORT modifiable_command_graph {
   /// @return Node added to the graph.
   node addImpl(const std::vector<node> &Dep);
 
+  // Adds all graph leaves as dependencies to the current node
+  void dependsOnLeaves(node Node);
```

Comment:
Preference for calling this `addGraphLeafDependencies` to make it clear that `Node` is being updated.